src/main/java/impl/expression/ExpressionParserImp.java

```diff
@@ -0,0 +1,38 @@
+package impl.expression;
+
+import api.expression.ExpressionParser;
+import api.expression.ParseException;
+
+public class ExpressionParserImp implements ExpressionParser {
+
+
+    public int parse(String s) throws ParseException {
+        if (s == null) {
+            throw new IllegalArgumentException("null input");
+        }
+        int expressionLength = s.length();
+        int currentNumber = 0;
+        int result = 0;
+        char previousSign = '+';
+        for (int i = 0; i < expressionLength; i++) {
+            char currentSymbol = s.charAt(i);
```

Comment:
As I see, you don't use `i` so you can use `for (char currentSymbol: s.toCharArray())` instead